polygeist/test/polygeist-opt/sycl/loop_internalization.mlir

```diff
@@ -16,27 +15,29 @@
-// CHECK:         %c0_i32 = arith.constant 0 : i32
-// CHECK-NEXT:    %c1_i32 = arith.constant 1 : i32
-// CHECK-NEXT:    [[TX:%.*]] = sycl.nd_item.get_global_id(%arg1, %c0_i32) : (memref<?x!sycl_nd_item_2_>, i32) -> i64  
+
+// COM: Get local ids:
+// CHECK:         [[LOCALID:%.*]] = sycl.local_id : !sycl_id_2_1
+// CHECK-NEXT:    [[LOCALID_:%.*]] = memref.alloca() : memref<1x!sycl_id_2_1>
+// CHECK:         memref.store [[LOCALID]], [[LOCALID_:%.*]][%c0{{.*}}] : memref<1x!sycl_id_2_1>
+// CHECK:         [[LOCALID0_:%.*]] = sycl.id.get [[LOCALID_]][%c0{{.*}}] : (memref<1x!sycl_id_2_1>, i32) -> memref<?xindex>
+// CHECK-NEXT:    [[LOCALID0:%.*]] = memref.load [[LOCALID0_]][%c0{{.*}}] : memref<?xindex>
+// CHECK:         [[LOCALID1_:%.*]] = sycl.id.get [[LOCALID_]][%c1{{.*}}] : (memref<1x!sycl_id_2_1>, i32) -> memref<?xindex>
+// CHECK-NEXT:    [[LOCALID1:%.*]] = memref.load [[LOCALID1_]][%c0{{.*}}] : memref<?xindex>
+
+// CHECK:         [[TX:%.*]] = sycl.nd_item.get_global_id(%arg1, %c0{{.*}}) : (memref<?x!sycl_nd_item_2_>, i32) -> i64  
+
+// COM: Get pointer to local memory:
 // CHECK-NEXT:    [[GETGLOBAL:%.*]] = memref.get_global @WGLocalMem : memref<64xi8, #sycl.access.address_space<local>>
-// SIZE1-NEXT:    [[TILESIZE:%.*]] = arith.constant 1 : index
-// SIZE2-NEXT:    [[TILESIZE:%.*]] = arith.constant 2 : index
+
+// CHECK-NEXT:    [[TILESIZE:%.*]] = arith.constant 2 : index
 // CHECK-NEXT:    affine.for [[IV1:%.*]] = 0 to [[MAP1]]()[[[TILESIZE]]] {
-// CHECK-NEXT:      %c0 = arith.constant 0 : index
-// CHECK-NEXT:      [[VIEW0:%.*]] = memref.view [[GETGLOBAL]][%c0][] : memref<64xi8, #sycl.access.address_space<local>> to memref<4x2xf32, #sycl.access.address_space<local>>
-// CHECK-NEXT:      %c32 = arith.constant 32 : index
-// CHECK-NEXT:      [[VIEW1:%.*]] = memref.view [[GETGLOBAL]][%c32][] : memref<64xi8, #sycl.access.address_space<local>> to memref<4x2xf32, #sycl.access.address_space<local>>
-// CHECK-NEXT:      spirv.ControlBarrier <Workgroup>, <Workgroup>, <SequentiallyConsistent|WorkgroupMemory>
+
 // CHECK-NEXT:      affine.for [[IV2:%.*]] = [[MAP2]]([[IV1]])[[[TILESIZE]]] to min [[MAP3]]([[IV1]])[[[TILESIZE]]] {
 // CHECK-NEXT:        [[IV2_CAST:%.*]] = arith.index_cast [[IV2]] : index to i64 
-// CHECK-NEXT:        sycl.constructor @id([[ID:%.*]], [[TX]], [[IV2_CAST]]) {{.*}} : (memref<?x!sycl_id_2_>, i64, i64)
-// CHECK-NEXT:        [[SUBSCR1:%.*]] = sycl.accessor.subscript %arg0[[[ID]]] : (memref<?x!sycl_accessor_2_f32_r_gb>, memref<?x!sycl_id_2_>) -> memref<?xf32>
-// CHECK-NEXT:        {{.*}} = affine.load [[SUBSCR1]][0] : memref<?xf32>
-// CHECK-NEXT:        [[SUBSCR2:%.*]] = sycl.accessor.subscript %arg0[[[ID]]] : (memref<?x!sycl_accessor_2_f32_r_gb>, memref<?x!sycl_id_2_>) -> memref<?xf32>
-// CHECK-NEXT:        {{.*}} = affine.load [[SUBSCR2]][0] : memref<?xf32>
+// CHECK:             affine.load
```

Comment:
Added early exit for affine loop, the CHECKs for the two affine loops will be changed when support for affine loops is added.